as1/src/main/java/socs/network/node/Router.java

```diff
@@ -165,55 +168,134 @@ public void run() {
                 if (inPacket.sospfType == 1) {
                   ports[routerIndex].router2.status = RouterStatus.TWO_WAY;
                   try {
+                    // this point is where the attached router confirms a two way connection
+                    // Update Link State Database
+                    LSA lsa = lsd._store.get(rd.simulatedIPAddress);
+                    // link state id is the simulated IP address of the router that the LSA is originating from
+                    // in this case it would be the first router
+                    //lsa.linkStateID = ports[routerIndex].router1.simulatedIPAddress;
+                    lsa.lsaSeqNumber += 1 ;
+                    LinkDescription ld = new LinkDescription();
+                    // link ID is the simulated ip address of the router the original is connected to
+                    ld.linkID = ports[routerIndex].router2.simulatedIPAddress;
+                    ld.portNum = inPacket.srcProcessPort;
+                    // weight of edge to other router?
+                    ld.tosMetrics = 999;
+                    lsa.links.add(ld);
+                    System.out.println("LSA from start:" + "\n" + lsa.toString());
+                    lsd.store(lsa);
+                    System.out.println("LSD from start:" + "\n" + lsd.toString());
+
+
                     // Need to send response
                     SOSPFPacket outPacket = new SOSPFPacket();
                     outPacket.srcProcessIP = inPacket.srcProcessIP; // localhost
                     outPacket.srcProcessPort = rd.processPortNumber;
                     outPacket.dstProcessPort = inPacket.srcProcessPort;
                     outPacket.srcIP = rd.simulatedIPAddress;
                     outPacket.dstIP = ports[routerIndex].router2.processIPAddress;
+
+
+                    //
+                    outPacket.linkdb = lsd;
+                    // might be removed later
+                    //System.out.println("troubleshooting");
+
+                    //outPacket.lsaArray.add(lsa);
                     outPacket.sospfType = 2; // We are sending the third handshake, ie. 2
                     outPacket.printPacket("Outgoing");
                     oos.writeObject(outPacket);
 
                     // Socket connection thread must stay alive
                     sequenceConcluded = false;
 
-                    // Update Link State Database
-                    LSA lsa = new LSA();
-                    lsa.linkStateID = ports[routerIndex].router2.simulatedIPAddress;
-                    lsa.lsaSeqNumber = Integer.MIN_VALUE;
+
+                  }
+                  catch (Exception e) {
+                    System.out.println(e);
+
+                  }
+                }
+
+                // Incoming 2 packet --> outgoing 3 packet
```

Comment:
This protocol should stop at packet 2, this is just necessary for the start function. Sending a packet with the header 3 for linstateDatabase swapping should be from a different function call other than start(). We want to be able to update our routing tables without calling start() everytime as well.